ivshmem/Device.c

```diff
@@ -90,8 +90,8 @@ NTSTATUS IVSHMEMEvtDevicePrepareHardware(_In_ WDFDEVICE Device, _In_ WDFCMRESLIS
 
 	if (deviceContext->interruptCount > 0)
 	{
-		deviceContext->interrupts = (WDFINTERRUPT*)MmAllocateNonCachedMemory(
-			sizeof(WDFINTERRUPT) * deviceContext->interruptCount);
+		deviceContext->interrupts = (WDFINTERRUPT*)ExAllocatePoolWithTag(NonPagedPool,
```

Comment:
Also need to fix the free side: ExFreePoolWithTag instead of MmFreeNonCachedMemory.

Reply:
thanks, done.